routers/api/v1/repo/release.go

```diff
@@ -246,6 +246,8 @@ func CreateRelease(ctx *context.APIContext) {
 		if err := release_service.CreateRelease(ctx.Repo.GitRepo, rel, nil, ""); err != nil {
 			if repo_model.IsErrReleaseAlreadyExist(err) {
 				ctx.Error(http.StatusConflict, "ReleaseAlreadyExist", err)
+			} else if models.IsErrProtectedTagName(err) {
+				ctx.Error(http.StatusMethodNotAllowed, "ProtectedTagName", err)
```

Comment:
Woudn't https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403 be more suitable?

Reply:
If we follow [GH API](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release), it should be 422 Conflict:

<img width="649" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/f12d1c55-a620-4025-9a42-9cac2dcf9b70">

Reply:
Yes, another status code is better. Just used the one all other places use.

Reply:
Not sure if it's wise to diverge from GH API, to keep things like actions compatible.